test/recipes/90-test_store.t

```diff
@@ -93,7 +93,11 @@ my @noexist_file_files =
 # @methods is a collection of extra 'openssl storeutl' arguments used to
 # try the different methods.
 my @methods;
-push @methods, [qw(-provider default -provider legacy)];
+if (disabled('legacy')) {
+    push @methods, [qw(-provider default)];
+} else {
+    push @methods, [qw(-provider default -provider legacy)];
+}
```

Comment:
To avoid repeats:

``` perl
push @methods, [qw(-provider default)];
push @methods, [qw(-provider legacy)] unless disabled('legacy');
```

Reply:
That's what I had originally, but for some reason it wasn't working.

Reply:
Oh, of course...  It's not a new method, it's another option on the *same* method.

``` perl
my @prov_method = qw(-provider default);
push @prov_method, qw(-provider legacy) unless disabled('legacy');
push @methods, [ @prov_method ];
```

Reply:
Fixed.